test/extension/handlers/topologymutation/handler.go

```diff
@@ -19,63 +19,225 @@ package topologymutation
 			infrav1.GroupVersion,
+			controlplanev1.GroupVersion,
+			bootstrapv1.GroupVersion,
 		),
 	}
 }
 
+var cgroupfs = "cgroupfs"
+
 // Handler is a topology mutation handler.
 type Handler struct {
 	decoder runtime.Decoder
 }
 
+var logger logr.Logger
+
 // GeneratePatches returns a function that generates patches for the given request.
 func (h *Handler) GeneratePatches(ctx context.Context, req *runtimehooksv1.GeneratePatchesRequest, resp *runtimehooksv1.GeneratePatchesResponse) {
 	log := ctrl.LoggerFrom(ctx)
+
+	logger = ctrl.LoggerFrom(ctx)
 	log.Info("GeneratePatches called")
 
 	walkTemplates(h.decoder, req, resp, func(obj runtime.Object, variables map[string]apiextensionsv1.JSON) error {
+		log.Info("Patching template", "template", obj.GetObjectKind().GroupVersionKind().String())
+
 		if dockerClusterTemplate, ok := obj.(*infrav1.DockerClusterTemplate); ok {
-			if err := patchDockerClusterTemplate(dockerClusterTemplate, variables); err != nil {
-				return err
+			if err := patchDockerClusterTemplate(ctx, dockerClusterTemplate, variables); err != nil {
+				log.Error(err, "error patching DockerClusterTemplate")
+				return errors.Wrap(err, "error patching DockerClusterTemplate")
+			}
+		}
+		if controlPlaneTemplate, ok := obj.(*controlplanev1.KubeadmControlPlaneTemplate); ok {
+			if err := patchKubeadmControlPlaneTemplate(ctx, controlPlaneTemplate, variables); err != nil {
+				log.Error(err, "error patching KubeadmControlPlaneTemplateß")
+				return errors.Wrapf(err, "error patching KubeadmControlPlaneTemplate")
+			}
+		}
+		if machineDeploymentBootstrapTemplate, ok := obj.(*bootstrapv1.KubeadmConfigTemplate); ok {
+			if err := patchMachineDeploymentBootstrapTemplate(ctx, machineDeploymentBootstrapTemplate, variables); err != nil {
+				log.Error(err, "error patching KubeadmConfigTemplate")
+				return errors.Wrap(err, "error patching KubeadmConfigTemplate")
+			}
+		}
+		if dockerMachineTemplate, ok := obj.(*infrav1.DockerMachineTemplate); ok {
+			if err := patchDockerMachineTemplate(ctx, dockerMachineTemplate, variables); err != nil {
+				log.Error(err, "error patching DockerMachineTemplate")
+				return errors.Wrap(err, "error patching DockerMachineTemplate")
 			}
 		}
-
 		return nil
 	})
 }
 
-// patchDockerClusterTemplate patches the DockerClusterTepmlate.
-func patchDockerClusterTemplate(dockerClusterTemplate *infrav1.DockerClusterTemplate, templateVariables map[string]apiextensionsv1.JSON) error {
-	// Get the variable value as JSON string.
-	value, err := patchvariables.GetVariableValue(templateVariables, "lbImageRepository")
+func patchKubeadmControlPlaneTemplate(ctx context.Context, kubeadmControlPlaneTemplate *controlplanev1.KubeadmControlPlaneTemplate, templateVariables map[string]apiextensionsv1.JSON) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	// 1) Patch .Spec.KubeadmConfigSpec.ClusterConfiguration.ImageRepository with the value from the Cluster Topology variables.
```

Comment:
Consider dropping imageRepo (like on Christian's PR)

But we can also keep it. We don't have to keep the ClusterClasses 100% sync